packages/-ember-data/tests/test-helper.js

```diff
@@ -3,41 +3,60 @@ import config from '../config/environment';
 import RSVP from 'rsvp';
 import { setApplication } from '@ember/test-helpers';
 import { start } from 'ember-qunit';
+import { DEBUG } from '@glimmer/env';
 
 import QUnit from 'qunit';
-import DS from 'ember-data';
 import { wait, asyncEqual, invokeAsync } from 'dummy/tests/helpers/async';
 import configureAsserts from 'dummy/tests/helpers/qunit-asserts';
+import { SHOULD_ASSERT_ALL } from './helpers/deprecated-test';
 
 configureAsserts();
 
 setApplication(Application.create(config.APP));
 
 const { assert } = QUnit;
-const transforms = {
-  boolean: DS.BooleanTransform.create(),
-  date: DS.DateTransform.create(),
-  number: DS.NumberTransform.create(),
-  string: DS.StringTransform.create(),
-};
 
 QUnit.begin(() => {
+  function assertAllDeprecations(assert) {
+    if (typeof assert.test.expected === 'number') {
+      assert.test.expected += 1;
+    }
+    assert.expectNoDeprecation(undefined, undefined, deprecation => {
+      // only assert EmberData deprecations
+      const id = deprecation.options.id;
+      const isEmberDataDeprecation = id.includes('DS') || id.includes('EmberData') || id.includes('ember-data');
+
+      if (!isEmberDataDeprecation) {
+        // eslint-disable-next-line no-console
+        console.log('Detected Non-Ember-Data Deprecation', deprecation);
+      }
+
+      return isEmberDataDeprecation;
+    });
+  }
+  // ensure we don't regress quietly
+  // this plays nicely with `expectDeprecation`
+  if (DEBUG) {
+    QUnit.config.modules.forEach(mod => {
```

Comment:
I don't _particularly_ like this bit, I'd rather add an import that does what we want `setupDeprecationAssertions(hooks)`. I think this is fine for a starting point, but would you mind adding a issue to move to a "less magic" setup (I'm happy to work on a small codemod to make it easy, since it seems like we have a few different things we want to tweak coming up).

Reply:
@rwjblue I don't know that I agree: manually adding this to every single test (or module) when you run it on every test seems annoying

Reply:
(but we should discuss)